cucumber/authenticators_azure/features/authn_azure_hosts.feature

```diff
@@ -0,0 +1,138 @@
+Feature: Azure Authenticator - Hosts can authenticate with Azure authenticator
+
+  In this feature we define an Azure authenticator in policy, define different
+  hosts and perform authentication with Conjur.
+
+  Background:
+    Given a policy:
+    """
+    - !policy
+      id: conjur/authn-azure/prod
+      body:
+      - !webservice
+
+      - !variable
+        id: provider-uri
+
+      - !group apps
+
+      - !permit
+        role: !group apps
+        privilege: [ read, authenticate ]
+        resource: !webservice
+    """
+    And I am the super-user
+    And I successfully set Azure variables
+
+  Scenario: Host with user-assigned-identity annotation is authorized
+    And I have host "user-assigned-identity-app"
+    And I set subscription-id annotation to host "user-assigned-identity-app"
+    And I set resource-group annotation to host "user-assigned-identity-app"
+    And I set user-assigned-identity annotation to host "user-assigned-identity-app"
+    And I grant group "conjur/authn-azure/prod/apps" to host "user-assigned-identity-app"
+    And I fetch a user-assigned Azure access token from inside machine
+    When I authenticate via Azure with token as host "user-assigned-identity-app"
+    Then host "user-assigned-identity-app" has been authorized by Conjur
+
+  Scenario: Host without resource-group annotation is denied
+    And I have host "no-resource-group-app"
+    And I set subscription-id annotation to host "no-resource-group-app"
+    And I grant group "conjur/authn-azure/prod/apps" to host "no-resource-group-app"
+    And I fetch an Azure access token from inside machine
+    And I save my place in the log file
+    When I authenticate via Azure with token as host "no-resource-group-app"
+    Then it is unauthorized
+    And The following appears in the log after my savepoint:
+    """
+    Errors::Authentication::AuthnAzure::RoleMissingConstraint
+    """
+
+  Scenario: Host without subscription-id annotation is denied
+    And I have host "no-subscription-id-app"
+    And I set resource-group annotation to host "no-subscription-id-app"
+    And I grant group "conjur/authn-azure/prod/apps" to host "no-subscription-id-app"
+    And I fetch an Azure access token from inside machine
+    And I save my place in the log file
+    When I authenticate via Azure with token as host "no-subscription-id-app"
+    Then it is unauthorized
+    And The following appears in the log after my savepoint:
+    """
+    Errors::Authentication::AuthnAzure::RoleMissingConstraint
```

Comment:
I think you need to add to log verification 
`Role does not have the required constraint: subscription_id
` because you are testing specific missing.